core/integration/module_test.go

```diff
@@ -2888,6 +2888,49 @@ class Foo {
 	}
 }
 
+func TestModuleScalarType(t *testing.T) {
+	// Verify use of a core scalar as an argument type.
+
+	t.Parallel()
+
+	type testCase struct {
+		sdk    string
+		source string
+	}
+	for _, tc := range []testCase{
+		{
+			sdk: "go",
+			source: `package main
+
+type Foo struct{}
+
+func (m *Foo) SayHello(platform Platform) string {
+	return "hello " + string(platform)
+}`,
+		},
+	} {
+		tc := tc
+
+		t.Run(tc.sdk, func(t *testing.T) {
+			t.Parallel()
+			c, ctx := connect(t)
+
+			modGen := c.Container().From(golangImage).
+				WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+				WithWorkdir("/work").
+				With(daggerExec("init", "--name=foo", "--sdk="+tc.sdk)).
+				With(sdkSource(tc.sdk, tc.source))
+
+			out, err := modGen.With(daggerQuery(`{foo{sayHello(platform: "linux/amd64")}}`)).Stdout(ctx)
+			require.NoError(t, err)
+			require.Equal(t, "hello linux/amd64", gjson.Get(out, "foo.sayHello").String())
+
+			_, err = modGen.With(daggerQuery(`{foo{sayHello(platform: "invalid")}}`)).Stdout(ctx)
+			require.ErrorContains(t, err, "unknown operating system or architecture")
```

Comment:
The error comes from the conversion in https://github.com/dagger/dagger/pull/7158/files#diff-1f747bfcbf3f060e72bd8621bc9daa68a4566fedc17f60fc837bf32d068c270bR222-R241.

Essentially, the failure comes from attempting to create a `Platform` scalar from a `string` input, which should fail, using the validation on the `Platform` scalar itself.

Reply:
Ok, so the test is basically validating between the GraphQL query and the API server, before it tries to send it as input to the SDK. But the SDK needs to register the argument correctly in order for the `Platform` scalar parsing to be picked up. 👍

What's strange now is that in Python, this query correctly fails with that error if I use `.Sync(ctx)` instead of `.Stdout(ctx)`, where the latter returns "hello invalid" if I look at the output. I would expect the same error regardless of `Stdout` or `Sync`. 🤔

Reply:
Uh oh, that's... not great. Out of curiosity, if you clean your cache, does this keep happening? I remember a specific cloud cache related issue when this could potentially happen, but otherwise not sure.

I'm assuming this is only for python? Or does go do it too?

Reply:
Only happens in Python it seems 😕

Reply:
I wiped everything and rebuilt. Now Python passes with `Stdout` too! Going to stress test this a bit more.

Reply:
Yeah, now it passes everytime no matter what I do. 🤷‍♂️